tests/datasource/fluent/test_spark_datasource.py

```diff
@@ -118,24 +117,22 @@ def test_spark_config_passed_to_execution_engine(
     persist,
     spark_session,
 ):
-    spark_config: SparkConfig | None = {
-        "spark.sql.catalogImplementation": "in-memory",
+    spark_config: SparkConfig = {
         "spark.app.name": "gx_spark_fluent_datasource_test",
         "spark.default.parallelism": 4,
-        "spark.driver.memory": "16g",
-        "spark.executor.memory": 471859200,
```

Comment:
Warnings are now raised for these config options because they can't be changed unless you restart the Spark Session. Options that raise warnings were either removed from tests or `with pytest.warns` was added.